reactor-core/src/main/java/reactor/core/publisher/FluxName.java

```diff
@@ -67,21 +66,21 @@
 		Objects.requireNonNull(tagName, "tagName");
 		Objects.requireNonNull(tagValue, "tagValue");
 
-		Set<Tuple2<String, String>> tags = Collections.singleton(Tuples.of(tagName, tagValue));
+		Map<String, String> tags = Collections.singletonMap(tagName, tagValue);
 
 		if (source instanceof FluxName) {
 			FluxName<T> s = (FluxName<T>) source;
 			if(s.tags != null) {
-				tags = new HashSet<>(tags);
-				tags.addAll(s.tags);
+				tags = new HashMap<>(s.tags);
```

Comment:
ideally, I guess it would make sense to use a `LinkedHashMap` to preserve insertion order, wdyt?